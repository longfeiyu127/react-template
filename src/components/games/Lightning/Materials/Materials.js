import React from 'react';
// import Tictactoe from './tictactoe'
// import Fivechess from './fivechess'
import './Materials.less';

class Materials extends React.Component {
  constructor() {
    super();
    this.state = {};
  }

  render() {
    // eslint-disable-next-line
    const { position, materialsType, size } = this.props;
    return (
      <div
        className="g-lig-materials"
        style={{
          top: `${position.top}rem`,
          left: `${position.left}rem`,
          height: `${size.height}rem`,
          width: `${size.width}rem`
        }}
      >
        {/* style={{fontWeight: (move === stepNumber) ? 900 : 500, cursor: 'pointer'}} */}
        <main className={['materials-main', materialsType].join(' ')} />
      </div>
    );
  }
}

export default Materials;
