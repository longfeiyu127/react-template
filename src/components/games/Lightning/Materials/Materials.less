
.g-lig-materials{
  position: absolute;
  >.materials-main {
    height: 100%;
    width: 100%;
  }
  // >.player{
  //   background: url('../../../../../assets/images/games/lightning/bullet/palyerBullet.png') no-repeat center/100% 100%;
  //   animation: flicker 100ms infinite;
  // }
  >.power{
    background: url('../../../../assets/images/games/lightning/materials/power_up_01.png') no-repeat center/100% 100%;
  }

  // @keyframes flicker {
  //   0%, 100% {
  //     opacity: 0;
  //   }
  //   50% {
  //     opacity: 1;
  //   }
  // }
}
