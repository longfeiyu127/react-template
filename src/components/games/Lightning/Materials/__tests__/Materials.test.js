import React from 'react';
import renderer from 'react-test-renderer';
import Materials from '../Materials';

describe('<Materials />', () => {
  it('component render', () => {
    const position = {
      top: 0.1,
      left: 0.1
    };
    const size = {
      height: 0.1,
      width: 0.1
    };
    let materials = renderer
      .create(
        <Materials position={position} size={size} materialsType="power" />
      )
      .toJSON();
    expect(materials).toMatchSnapshot();

    materials = renderer
      .create(
        <Materials position={position} size={size} materialsType="power" />
      )
      .toJSON();
    expect(materials).toMatchSnapshot();
  });
});
