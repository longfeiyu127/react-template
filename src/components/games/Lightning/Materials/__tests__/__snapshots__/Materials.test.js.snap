// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Materials /> component render 1`] = `
<div
  className="g-lig-materials"
  style={
    Object {
      "height": "0.1rem",
      "left": "0.1rem",
      "top": "0.1rem",
      "width": "0.1rem",
    }
  }
>
  <main
    className="materials-main power"
  />
</div>
`;

exports[`<Materials /> component render 2`] = `
<div
  className="g-lig-materials"
  style={
    Object {
      "height": "0.1rem",
      "left": "0.1rem",
      "top": "0.1rem",
      "width": "0.1rem",
    }
  }
>
  <main
    className="materials-main power"
  />
</div>
`;
